recipes/libidn/all/conanfile.py

```diff
@@ -0,0 +1,131 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class LibIdn(ConanFile):
+    name = "libidn"
+    description = "GNU Libidn is a fully documented implementation of the Stringprep, Punycode and IDNA 2003 specifications."
+    homepage = "https://www.gnu.org/software/libidn/"
+    topics = ("conan", "libidn", "encode", "decode", "internationalized", "domain", "name")
+    license = "GPL-3.0-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "threads": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "threads": True,
+    }
+    settings = "os", "arch", "compiler", "build_type"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows due to libtool limitation")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("libiconv/1.16")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("automake/1.16.2")
```

Comment:
Isn't it already available in msys2 package?

Reply:
automake is required for the cl and ar wrapper.
I don't know how to get this from the msys2 wrapper.

And besides, by using our own automake, we have more control.